app/controllers/lobbytrack_reports_controller.rb

```diff
@@ -9,32 +9,40 @@ def index
   def visits
     id = params[:lobbytrack_report][:visit_id]
 
-    begin
-      @lobbytrack_visitor = LobbytrackReport.query_visitor(id)
-      @lobbytrack_reports = LobbytrackReport.query_visits(id)
-    rescue TinyTds::Error => e
-      flash[:error] = e
+    @lobbytrack_form = LobbytrackReport.new(visit_id: id)
+    if @lobbytrack_form.valid?
+      begin
+        @lobbytrack_visitor = LobbytrackReport.query_visitor(id)
+        @lobbytrack_reports = LobbytrackReport.query_visits(id)
+      rescue TinyTds::Error => e
+        flash[:error] = e
+      end
     end
 
     return if @lobbytrack_reports&.any?
 
     flash[:warning] = "There is no attendance history for id #{id}"
+    flash[:error] = 'Invalid ID entered.'
```

Comment:
When I put in a valid ID, I get the "invalid ID entered" message instead of the "unable to connect" message.

Reply:
This was in the wrong place, should've been in the `else` condition above it.